packages/flutter_tools/lib/src/build_system/targets/dart_plugin_registrant.dart

```diff
@@ -45,9 +45,9 @@ class DartPluginRegistrantTarget extends Target {
     final String targetFile = environment.defines[kTargetFile] ??
         environment.fileSystem.path.join('lib', 'main.dart');
     final File mainFile = environment.fileSystem.file(targetFile);
-    final Uri mainFileUri = mainFile.uri;
-    assert(packagesFile.path != null);
+    final Uri mainFileUri = mainFile.absolute.uri;
     final String mainUri = packageConfig.toPackageUri(mainFileUri)?.toString();
```

Comment:
What if the main file URI is not in any package?

Reply:
i.e. `flutter run --target some/absolute/path/test/test_foo.dart`

Reply:
What would be an example?

Reply:
Do we validate the target earlier?

Reply:
Thats not an invalid target though - you'd need to fallback to embedding an absolute URI

Reply:
sure, but would it make sense to throw an error if package config says that such entrypoint doesn't exist?

Reply:
I'm not quite sure what you mean - if the package config fails to resolve the absolute file URI to a package URI, it only means that the file was not under any of the defined package roots. Its still an otherwise valid target file

Reply:
ah I see what you mean, `import 'package:<package-name>/some/absolute/path/test/test_foo.dart` in this case, right?

Reply:
For example:

```dart
import 'file:///C:/Users/Jonah/flutter/packages/flutter_tools/bin/flutter_tools.dart';
```

Reply:
This is totally valid (well, on my computer at least)

Reply:
👍 I cleaned up the test file a bit, and added the new test. PTAL